bigquery_etl/public_data/publish_json.py

```diff
@@ -114,6 +109,12 @@ def publish_json(self):
 
         self._publish_last_updated()
 
+        # delete temporary artifacts
+        if self.temp_table:
+            self.client.delete_table(self.temp_table)
+
+        self._clear_stage_directory()
+
```

Comment:
I moved the deletion of the temporary tables and files to the end of the script here.

Reply:
this needs to occur in a `try ... finally` block to ensure that resources are deleted even when an error occurs.

Reply:
Right, should be fixed now.